### PR TITLE
add rpc

### PR DESCRIPTION
**Pull request by prin-r**

### CODE REVIEW

File: go/eth/rpc.go [hunk 1]
```diff
@@ -13,29 +20,68 @@ type Signature struct {
...
+	}
+	return result, nil
 }
 
 // SignMessage returns the signature of signing the given message using Ethereum's message
 // signing scheme.
 func SignMessage(message []byte) (Signature, error) {
 	// TODO(prin-r): Implement this
-	panic("SignMessage: not implemented")
+	if client == nil {
```
Comment (taobun) [inline go/eth/rpc.go]: We need client to sign a message?

File: go/eth/rpc.go [hunk 2]
```diff
@@ -13,29 +20,68 @@ type Signature struct {
...
-func GetStorageAt(contract common.Address, location common.Hash) (common.Hash, error) {
+func GetStorageAt(contract common.Address, location common.Hash) ([]byte, error) {
 	// TODO(prin-r): Implement this
-	panic("GetStorageAt: not implemented")
+	if client == nil {
+		return []byte{}, errors.New("Initialization is required")
+	}
+	result, err := client.StorageAt(context.Background(), contract, location, nil)
+	if err != nil {
+		return []byte{}, errors.New("no private key found")
```
Comment (taobun) [inline go/eth/rpc.go]: Error msg seem wrong

File: go/eth/rpc.go [hunk 3]
```diff
@@ -13,29 +20,68 @@ type Signature struct {
 }
 
 // TODO(prin-r): Declare unexported variable that represents a connection to an Ethereum node
+var client *ethclient.Client
+var pk *ecdsa.PrivateKey
 
 func init() {
 	// TODO(prin-r): Initialize a connection to Ethereum
+	var err error
+	eth_rpc := os.Getenv("ETH_RPC")
```
Comment (sorawit) [inline go/eth/rpc.go]: NO_UNDERSCORE_PLEASE

File: go/eth/rpc.go [hunk 4]
```diff
@@ -13,29 +20,68 @@ type Signature struct {
...
+	client, err = ethclient.Dial(eth_rpc)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	pk, err = crypto.HexToECDSA(os.Getenv("PK"))
```
Comment (sorawit) [inline go/eth/rpc.go]: If you are not planning to implement the no env case here. Please add `// TODO(prin-r): something`

File: go/eth/rpc.go [hunk 5]
```diff
@@ -13,29 +20,68 @@ type Signature struct {
...
+	if pk == nil {
+		return common.Address{}, errors.New("no private key found")
+	}
+	return crypto.PubkeyToAddress(pk.PublicKey), nil
 }
 
 // GetStorageAt returns the value at the given storage location of the given Ethereum
 // contract address.
-func GetStorageAt(contract common.Address, location common.Hash) (common.Hash, error) {
+func GetStorageAt(contract common.Address, location common.Hash) ([]byte, error) {
```
Comment (sorawit) [inline go/eth/rpc.go]: Let's not change this

File: go/eth/rpc.go (hunk 4 above)
Comment (sorawit) [inline go/eth/rpc.go]: Let's name this variable `ETH_PRIVATE_KEY`

**Pull request merged**